dig.go

```diff
@@ -692,6 +692,36 @@ func isFieldOptional(f reflect.StructField) (bool, error) {
 	return optional, err
 }
 
+type group struct {
+	name    string
+	flatten bool
+}
+
+func parseGroup(f reflect.StructField) (group, error) {
+	tag := f.Tag.Get(_groupTag)
+	if tag == "" {
+		return group{}, nil
```

Comment:
Looking at its usage, there's no valid scenario for parseGroup where the tag
isn't set. Should we panic since that's a bug?